webui/react/src/hooks/useModal/UserSettings/useModalPasswordChange.tsx

```diff
@@ -1,146 +1,139 @@
-import { Button, Form, Input, message } from 'antd';
-import React, { useCallback, useState } from 'react';
+import { Form, Input, message } from 'antd';
+import { FormInstance } from 'antd/lib/form/hooks/useForm';
+import React, { useCallback } from 'react';
 
 import { useStore } from 'contexts/Store';
 import { login, setUserPassword } from 'services/api';
+import { ErrorType } from 'shared/utils/error';
 import handleError from 'utils/error';
 
 import useModal, { ModalHooks } from '../useModal';
 
-import css from './useModalPasswordChange.module.scss';
-
 interface Props {
-  onComplete: () => void;
+  form: FormInstance;
+  username?: string;
 }
 
-const ChangePassword: React.FC<Props> = ({ onComplete }) => {
-  const { auth } = useStore();
-  const username = auth.user?.username ?? '';
-  const userId = auth.user?.id ?? 0;
+export const MODAL_HEADER_LABEL = 'Change Password';
+export const OLD_PASSWORD_LABEL = 'Old Password';
+export const OLD_PASSWORD_NAME = 'oldPassword';
+export const NEW_PASSWORD_LABEL = 'New Password';
+export const NEW_PASSWORD_NAME = 'newPassword';
+export const CONFIRM_PASSWORD_LABEL = 'Confirm Password';
+export const CONFIRM_PASSWORD_NAME = 'confirmPassword';
+export const CANCEL_BUTTON_LABEL = 'Cancel';
+export const OK_BUTTON_LABEL = 'Change Password';
+export const INCORRECT_PASSWORD_MESSAGE = 'Incorrect password.';
+export const NEW_PASSWORD_REQUIRED_MESSAGE = 'New password required.';
+export const PASSWORD_TOO_SHORT_MESSAGE = 'Password isn\'t long enough.';
+export const PASSWORD_UPPERCASE_MESSAGE = 'Password must include a uppercase letter.';
+export const PASSWORD_LOWERCASE_MESSAGE = 'Password must include a lowercase letter.';
+export const PASSWORD_NUMBER_MESSAGE = 'Password must include a number.';
+export const CONFIRM_PASSWORD_REQUIRED_MESSAGE = 'Confirmed password required.';
+export const PASSWORDS_NOT_MATCHING_MESSAGE = 'Passwords do not match.';
+export const API_SUCCESS_MESSAGE = 'Password updated.';
+export const API_ERROR_MESSAGE = 'Could not update password.';
+
+const ModalForm: React.FC<Props> = ({ form, username = '' }) => (
+  <Form form={form} layout="vertical">
+    <Form.Item
+      label={OLD_PASSWORD_LABEL}
+      name={OLD_PASSWORD_NAME}
+      required
+      rules={[
+        {
+          message: INCORRECT_PASSWORD_MESSAGE,
+          validator: async (rule, value) => {
+            await login({ password: value ?? '', username });
+          },
+        },
+      ]}
+      validateTrigger={[ 'onSubmit' ]}>
```

Comment:
realized this may not be necessary since we're not using the submit action for the form anymore.